### PR TITLE
Remote Linked Reports

### PR DESCRIPTION
**Pull request by proteusvacuum**
## Summary

Adds the ability to remote link reports, and use these in linked apps. 
https://dimagi-dev.atlassian.net/browse/SC-1475

This became bigger than initially expected because remotely linking reports wasn't working as expected. I've added a new tab on the linked project space page that only shows if there are remote linked reports to actually pull. Once these are linked, they can then be updated through the regular channel. Feedback welcome on that UI. 

Note, this is a long demo, but shows all the moving parts here. In order:
- Add new report to the upstream app
- Pull the upstream app, but get an error since that report doesn't exist downstream
- Pull that report downstream
- Pull the app again successfully
- Load the downstream app and see the report successfully added with data from the downstream domain
Remote Reports

## Feature Flag

Linked domains
Mobile UCR

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js [hunk 1]
```diff
@@ -75,6 +75,29 @@ hqDefine("linked_domain/js/domain_links", [
             return DomainLink(link);
         }));
 
+        self.linkable_ucr = ko.observableArray(_.map(data.linkable_ucr, function (report) {
```
Comment (gherceg) [inline corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:78]: nit: use js naming scheme `linkableUcr`

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js [hunk 2]
```diff
@@ -75,6 +75,29 @@ hqDefine("linked_domain/js/domain_links", [
...
+            return RemoteLinkableReport(report, self.master_link);
+        }));
+        self.createRemoteReportLink = function (reportId) {
+            _private.RMI("create_remote_report_link", {
+                "master_domain": self.master_link.master_domain,
+                "linked_domain": self.master_link.linked_domain,
```
Comment (gherceg) [inline corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:84]: there will be an overhaul in the future, but if it isn't too much of a pain to change `upstream_domain` and `downstream_domain` that would be ideal.

File: corehq/apps/linked_domain/static/linked_domain/js/domain_links.js [hunk 3]
```diff
@@ -111,6 +134,36 @@ hqDefine("linked_domain/js/domain_links", [
...
+    var RemoteLinkableReport = function (report, masterLink) {
+        var self = {};
+        self.id = report.id;
+        self.title = report.title;
+        self.alreadyLinked = ko.observable(report.already_linked);
+
+        self.createLink = function () {
+            _private.RMI("create_remote_report_link", {
+                "master_domain": masterLink.master_domain,
+                "linked_domain": masterLink.linked_domain,
```
Comment (gherceg) [inline corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:146]: same here

File: corehq/apps/linked_domain/templates/linked_domain/partials/link_remote_ucr.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+
+<table class="table table-striped table-hover">
+  <thead>
+    <tr>
+      <th>{% trans "Report Name" %}</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody data-bind="foreach: linkable_ucr">
```
Comment (gherceg) [inline corehq/apps/linked_domain/templates/linked_domain/partials/link_remote_ucr.html:12]: would need to update to linkableUcr if updating to js naming scheme

**Review by gherceg — APPROVED** [review]
Looks good. Just left some minor nitpicky comments. Thanks for adding the new tab in it's own file.

**Pull request merged**